pkg/server/server.go

```diff
@@ -177,10 +186,41 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
 	c.GenericConfig.RequestInfoResolver = requestinfo.NewFactory() // must be set here early to avoid a crash in the EnableMultiCluster roundtrip wrapper
 
 	// Setup kcp * informers, but those will need the identities for the APIExports used to make the APIs available.
-	// The identities are not known before we can get them from the APIExports via the loopback client, hence we postpone
-	// this to getOrCreateKcpIdentities() in the kcp-start-informers post-start hook.
+	// The identities are not known before we can get them from the APIExports via the loopback client or from the root shard in case this is a non-root shard,
+	// hence we postpone this to getOrCreateKcpIdentities() in the kcp-start-informers post-start hook.
 	// The informers here are not  used before the informers are actually started (i.e. no race).
-	nonIdentityKcpClusterClient, err := kcpclient.NewClusterForConfig(c.GenericConfig.LoopbackClientConfig) // can only used for wildcard requests of apis.kcp.dev
+
+	nonIdentityKcpClusterConfig := c.GenericConfig.LoopbackClientConfig
+	if len(c.Options.Extra.RootShardKubeconfigFile) > 0 {
+		// TODO(p0lyn0mial): use kcp-admin instead of system:admin
+		nonIdentityRootKcpShardSystemAdminConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: c.Options.Extra.RootShardKubeconfigFile}, &clientcmd.ConfigOverrides{CurrentContext: "system:admin"}).ClientConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to load the kubeconfig from: %s, for the root shard, err: %w", c.Options.Extra.RootShardKubeconfigFile, err)
+		}
+		nonIdentityKcpClusterConfig = nonIdentityRootKcpShardSystemAdminConfig                                        // assign to nonIdentityKcpClusterConfig to the root shard
+		nonIdentityRootKcpShardClient, err := kcpclient.NewClusterForConfig(nonIdentityRootKcpShardSystemAdminConfig) // can only used for wildcard requests of apis.kcp.dev
+		if err != nil {
+			return nil, err
+		}
+		var rootKcpShardIdentityConfig *rest.Config
+		rootKcpShardIdentityConfig, c.ResolveRootKcpShardIdentities = boostrap.NewConfigWithWildcardIdentities(nonIdentityRootKcpShardSystemAdminConfig, boostrap.KcpRootGroupExportNames, boostrap.KcpRootGroupResourceExportNames, nonIdentityRootKcpShardClient.Cluster(tenancyv1alpha1.RootCluster))
```

Comment:
I am lost in this wiring. Don't we have these two cases:
1. *we are the root shard*. Then we need the old, staged flow where we start the local informer only for apis.kcp.dev first, then pick the identities from the APIExports, inject them into the roundtripper and then start the rest.
2. *we are NOT the root shard*. Then we can run `resolveRootKcpShardIdentities` here in-place very early before anything of the code further down runs, i.e. before any of the informers have been started.